types/katex.d.ts

```diff
@@ -20,6 +20,43 @@ export type TrustContext =
     | { command: "\\htmlStyle", style: string }
     | { command: "\\htmlData", attributes: Record<string, string> }
 
+
+type KatexCatcodes = Record<string, number>;
+
+interface KatexLexerInterFace {
+  input: string;
+  tokenRegex: RegExp;
+  settings: Required<KatexOptions>;
+  catcodes: KatexCatcodes;
+}
+
+interface KatexSourceLocation {
+  start: number;
+  end: number;
+  lexer: KatexLexerInterFace;
+}
+
+export interface KatexToken {
```

Comment:
BTW, to safely fit all users even they are patching katex for they own usage, we can let katexToken extends `Record<string, unknown>`, so other properties hacked by users can be added without causing any type issues.

Requesting for a feedback here.

Reply:
I don't think that's necessary; the user can always [merge interfaces](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces) to extend anything we provide. (You'd need to wrap it in `declare module "katex" { ... }` I believe.)